### ISSUE TITLE
Error parsing definition properties when adding "allOf"

### ISSUE THREAD

**Issue by mis101bird** [swagger-api/swagger-parser#396]
When I parse definition properties without "allOf", I can parse the property whose name is "properties" in AnalysisServicesServer.
```
	@Test
	public void definitionPropertiesIssue(){
	    String yaml =
                "swagger: '2.0'\n" +
                "info:\n" +
                "  title: test\n" +
                "  version: '0.0.0'\n" +
                "parameters:\n" +
                "  foo-id:\n" +
                "    name: id\n" +
                "    in: path\n" +
                "    type: string\n" +
                "    required: true\n" +
                "paths:\n" +
                "  /foos/{id}:\n" + 
                "    parameters:\n" +
                "        - $ref: '#/parameters/foo-id'\n" +
                "    get:\n" +
                "      responses:\n" +
                "        200:\n" +
                "          schema:\n" +
                "            $ref: '#/definitions/foo'\n" +
                "    put:\n" +
                "      parameters:\n" +
                "        - name: foo\n" +
                "          in: body\n" +
                "          required: true\n" +
                "          schema:\n" +
                "            $ref: '#/definitions/foo'\n" +
                "      responses:\n" +
                "        200:\n" +
                "          schema:\n" +
                "            $ref: '#/definitions/foo'\n" +
                "definitions:\n" +
                "  AnalysisServicesServer:\n"+
                "      description: Represents an instance of an Analysis Services resource.\n"+
                "      properties:\n"+
                "        properties:\n"+
                "          $ref: '#/definitions/AnalysisServicesServerProperties'\n"+
                "          description: Properties of the provision operation request.\n"+
                "      type: object";
        SwaggerParser parser = new SwaggerParser();
        Swagger swagger = parser.parse(yaml);
        Model model = (swagger.getDefinitions().get("AnalysisServicesServer"));
        Map<String, Property> properties = model.getProperties();
        assertEquals(properties.get("properties").getDescription(), "Properties of the provision operation request."); // Pass
	}
```
But when I add allOf, I get NullPointerException from model.getProperties()
```
        @Test
	public void definitionPropertiesIssue(){
	    String yaml =
                "swagger: '2.0'\n" +
                "info:\n" +
                "  title: test\n" +
                "  version: '0.0.0'\n" +
                "parameters:\n" +
                "  foo-id:\n" +
                "    name: id\n" +
                "    in: path\n" +
                "    type: string\n" +
                "    required: true\n" +
                "paths:\n" +
                "  /foos/{id}:\n" + 
                "    parameters:\n" +
                "        - $ref: '#/parameters/foo-id'\n" +
                "    get:\n" +
                "      responses:\n" +
                "        200:\n" +
                "          schema:\n" +
                "            $ref: '#/definitions/foo'\n" +
                "    put:\n" +
                "      parameters:\n" +
                "        - name: foo\n" +
                "          in: body\n" +
                "          required: true\n" +
                "          schema:\n" +
                "            $ref: '#/definitions/foo'\n" +
                "      responses:\n" +
                "        200:\n" +
                "          schema:\n" +
                "            $ref: '#/definitions/foo'\n" +
                "definitions:\n" +
                "  AnalysisServicesServer:\n"+
                "      allOf:\n"+  // Add allOf
                "          - $ref: '#/definitions/Resource'\n"+
                "      description: Represents an instance of an Analysis Services resource.\n"+
                "      properties:\n"+
                "        properties:\n"+
                "          $ref: '#/definitions/AnalysisServicesServerProperties'\n"+
                "          description: Properties of the provision operation request.\n"+
                "      type: object";
        SwaggerParser parser = new SwaggerParser();
        Swagger swagger = parser.parse(yaml);
        Model model = (swagger.getDefinitions().get("AnalysisServicesServer"));
        Map<String, Property> properties = model.getProperties();
        assertEquals(properties.get("properties").getDescription(), "Properties of the provision operation request."); //NullPointerException
	}
```
I guess it may be a bug. Is there any advices to avoid the bug being happened? 
Thanks a lot.

**Comment by GregBragg** [comment]
I am experiencing the same bug when parsing a swagger: '2.0' yaml definition file using the "allOf" element for a definition, any idea when this will be fixed?

**Comment by gracekarina** [comment]
@mis101bird Can you provide the full spec?

**Comment by Impakt** [comment]
There's a spec [here](https://raw.githubusercontent.com/APIs-guru/openapi-directory/master/APIs/azure.com/cognitiveservices-LUIS-Programmatic/v2.0/swagger.yaml) that reproduces the issue

**Comment by ununhexium** [comment]
Encountered the problem in 2.0.15.
Tried to do this ticket and realized that I can't reproduce it with 2.0.16.

Was probably fixed by 6979216d686b83ac19287e1f73d7d528c72788cd #1143

**Comment by gracekarina** [comment]
closed by #1143
